### PR TITLE
PORTALS-3052

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: packages/synapse-react-client/src/components/FullTextSearch.tsx [hunk 1]
```diff
@@ -36,23 +42,50 @@ export const FullTextSearch: React.FunctionComponent<FullTextSearchProps> = ({
...
-        }
-        // PORTALS-2093: does this additional filter already exist?
-        const found = additionalFilters.find(
-          filter =>
-            filter.concreteType == textMatchesQueryFilter.concreteType &&
-            filter.searchExpression == textMatchesQueryFilter.searchExpression,
-        )
-        if (found) {
-          return request
+        const idColumnModel = getColumnModel('id')
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/FullTextSearch.tsx:45]: Maybe use `getFileColumnModelId` (in SynapseTableUtils), unless that doesn't work for some known cases.

File: packages/synapse-react-client/src/components/FullTextSearch.tsx [hunk 2]
```diff
@@ -36,23 +42,50 @@ export const FullTextSearch: React.FunctionComponent<FullTextSearchProps> = ({
...
+        const isSynapseID = searchText.match(SYNAPSE_ENTITY_ID_REGEX)
+        if (
+          isSynapseID &&
+          idColumnModel &&
+          idColumnModel.columnType == ColumnTypeEnum.ENTITYID
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/FullTextSearch.tsx:50]: Can remove this last check if you use `getFileColumnModelId`

File: packages/synapse-react-client/src/components/FullTextSearch.tsx [hunk 3]
```diff
@@ -36,23 +42,50 @@ export const FullTextSearch: React.FunctionComponent<FullTextSearchProps> = ({
...
+        ) {
+          const singleValueQueryFilter: ColumnSingleValueQueryFilter = {
+            concreteType:
+              'org.sagebionetworks.repo.model.table.ColumnSingleValueQueryFilter',
+            columnName: 'id',
+            operator: ColumnSingleValueFilterOperator.IN,
+            values: [searchText],
+          }
+          const matchingFilter = additionalFilters.find(
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/FullTextSearch.tsx:59]: Nice touch to replace a filter with a different term on the same column, just want to a comment to make it clear what's happening
```suggestion
          // Replace the active filter on the column, if one exists
          const matchingFilter = additionalFilters.find(
```

**Review by nickgros — APPROVED** [review]

File: packages/synapse-react-client/src/components/FullTextSearch.tsx (hunk 2 above)
Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/FullTextSearch.tsx]: Done

File: packages/synapse-react-client/src/components/FullTextSearch.tsx (hunk 1 above)
Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/FullTextSearch.tsx]: Done

**Pull request merged**